### PR TITLE
feat: Auth form visual enhancements (master)

### PR DESCRIPTION
**Pull request by Lunyachek**
### Description

Cosmetic improvements for the login/registration page. Changes in React are related to the situation when third-party login providers is disabled, but when switching between Login and Registration tabs, we still see a preloader that is not necessary in this situation.

https://user-images.githubusercontent.com/19806032/222164313-f1e985d9-47e1-4b18-8b27-8280f5c6a7fd.mov

And sass changes are related to the suggested username field, which is too short and can be extended to match the neighboring inputs.

image1

After making our changes, we no longer see preloaders in cases where third-party login providers are turned off.

https://user-images.githubusercontent.com/19806032/222174870-e76fe894-1a63-448b-bc08-ffd9df8dca0b.mov

And suggested username field looks pretty

Снимок экрана 2023-03-01 в 16 46 17

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Lunyachek!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: src/login/LoginPage.jsx [hunk 1]
```diff
@@ -167,6 +167,7 @@ class LoginPage extends React.Component {
     const isInstitutionAuthActive = !!secondaryProviders.length && !currentProvider;
     const isSocialAuthActive = !!providers.length && !currentProvider;
     const isEnterpriseLoginDisabled = getConfig().DISABLE_ENTERPRISE_LOGIN;
+    const ThirdPartyAuthPreloader = isSocialAuthActive || (isEnterpriseLoginDisabled && isInstitutionAuthActive);
```
Comment (syedsajjadkazmii) [inline src/login/LoginPage.jsx:170]: I believe `ThirdPartyAuthPreloader` is supposed to be a boolean flag. Could you please refactor it to `thirdPartyAuthPreloader`

File: src/register/ThirdPartyAuth.jsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ const ThirdPartyAuth = (props) => {
   const isInstitutionAuthActive = !!secondaryProviders.length && !currentProvider;
   const isSocialAuthActive = !!providers.length && !currentProvider;
   const isEnterpriseLoginDisabled = getConfig().DISABLE_ENTERPRISE_LOGIN;
+  const ThirdPartyAuthPreloader = isSocialAuthActive || (isEnterpriseLoginDisabled && isInstitutionAuthActive);
```
Comment (syedsajjadkazmii) [inline src/register/ThirdPartyAuth.jsx:27]: [Same concern again] I believe ThirdPartyAuthPreloader is supposed to be a boolean flag. Could you please refactor it to thirdPartyAuthPreloader

File: src/sass/_style.scss [hunk 3]
```diff
@@ -512,6 +512,10 @@ select.form-control {
   }
 }
 
+.content {
```
Comment (syedsajjadkazmii) [inline src/sass/_style.scss:515]: Could you please point me to this class's usage?

File: src/login/LoginPage.jsx [hunk 4]
```diff
@@ -167,6 +167,7 @@ class LoginPage extends React.Component {
...
 
     return (
       <>
```
Comment (syedsajjadkazmii) [inline src/login/LoginPage.jsx:173]: The condition at line 174 is the same as used in `ThirdPartyAuthPreloader`. Can we use it at line 174 also? Maybe we can give it a name that works for both conditions here?

File: src/register/ThirdPartyAuth.jsx [hunk 5]
```diff
@@ -24,6 +24,7 @@ const ThirdPartyAuth = (props) => {
...
 
   return (
     <>
```
Comment (syedsajjadkazmii) [inline src/register/ThirdPartyAuth.jsx:30]: The comment regarding using the same variable for the heading condition also applies here.

**Review by syedsajjadkazmii — COMMENTED** [review]
Overall the layout is changed for me, I can see in your screenshots that it's fine but the left panel width is decreased for me. Although it's working fine on the master branch. Could you please also look into this?[
Screenshot 2023-03-08 at 10 20 39 AM
](url)

**Comment by mphilbrick211** [comment]
> Overall the layout is changed for me, I can see in your screenshots that it's fine but the left panel width is decreased for me. Although it's working fine on the master branch. Could you please also look into this?[ Screenshot 2023-03-08 at 10 20 39 AM ](url)

Hi @Lunyachek - friendly ping on this :)

**Review by syedsajjadkazmii — COMMENTED** [review]
One more thing that we noticed (thanks to @zainab-amir )

If we use this approach for hiding the social auth skeleton when they are inactive, then it will also hide the skeleton when social auth is active. Because we get providers in MFE_context api response and based on the response we render these buttons if providers are available.

**Comment by mphilbrick211** [comment]
Hi @Lunyachek! Looks like some tests need to be re-run and branch conflicts need to be resolved.

@zainab-amir @syedsajjadkazmii - Product has completed their review. Once the conflicts and tests are all set, please review and merge if all looks good. Thanks!

**Comment by mphilbrick211** [comment]
Hi @syedsajjadkazmii and @zainab-amir - any update on this?

**Comment by syedsajjadkazmii** [comment]
> Hi @syedsajjadkazmii and @zainab-amir - any update on this?

Hi @mphilbrick211, we are waiting on the author to resolve the suggestions before we proceed with this.

**Comment by mphilbrick211** [comment]
> > Hi @syedsajjadkazmii and @zainab-amir - any update on this?
> 
> Hi @mphilbrick211, we are waiting on the author to resolve the suggestions before we proceed with this.

@Lunyachek - flagging this for you :)

**Comment by mphilbrick211** [comment]
Hi @Lunyachek! Just checking to see if you're able to address the items above? Thanks!

**Comment by mphilbrick211** [comment]
> Hi @Lunyachek! Just checking to see if you're able to address the items above? Thanks!

Hi @Lunyachek - are you planning to pursue this pull request? Please let us know if you have any questions. Thanks!

File: src/login/LoginPage.jsx (hunk 1 above)
Comment (Lunyachek) [reply src/login/LoginPage.jsx]: I agree, changed to `isThirdPartyAuthActive`

File: src/login/LoginPage.jsx (hunk 4 above)
Comment (Lunyachek) [reply src/login/LoginPage.jsx:173]: Changed to isThirdPartyAuthActive

File: src/register/ThirdPartyAuth.jsx (hunk 2 above)
Comment (Lunyachek) [reply src/register/ThirdPartyAuth.jsx]: Changed to isThirdPartyAuthActive

File: src/sass/_style.scss (hunk 3 above)
Comment (Lunyachek) [reply src/sass/_style.scss]: This was related to neighboring PR about horizontal scroll and no valid for now. I removed this style

File: src/register/ThirdPartyAuth.jsx (hunk 5 above)
Comment (Lunyachek) [reply src/register/ThirdPartyAuth.jsx:30]: I have changes all `ThirdPartyAuthPreloader` to `isThirdPartyAuthActive`

**Comment by Lunyachek** [comment]
> Overall the layout is changed for me, I can see in your screenshots that it's fine but the left panel width is decreased for me. Although it's working fine on the master branch. Could you please also look into this?[ Screenshot 2023-03-08 at 10 20 39 AM ](url)

I removed changes in css related to the `suggested username field` because of fresh master branch already includes enhancements for this field. And left bar, shown on this screenshot, increased to it normal size

**Comment by mphilbrick211** [comment]
Thanks, @Lunyachek! It looks like there's a failing test - would you mind taking a look?

**Comment by syedsajjadkazmii** [comment]
> One more thing that we noticed (thanks to @zainab-amir )
> 
> If we use this approach for hiding the social auth skeleton when they are inactive, then it will also hide the skeleton when social auth is active. Because we get providers in MFE_context api response and based on the response we render these buttons if providers are available.

Hi @Lunyachek, could you please consider this scenario too?

**Comment by mphilbrick211** [comment]
> > One more thing that we noticed (thanks to @zainab-amir )
> > If we use this approach for hiding the social auth skeleton when they are inactive, then it will also hide the skeleton when social auth is active. Because we get providers in MFE_context api response and based on the response we render these buttons if providers are available.
> 
> Hi @Lunyachek, could you please consider this scenario too?

Hi @Lunyachek! Just checking to see if you plan to pursue this PR?

**Comment by Lunyachek** [comment]
> One more thing that we noticed (thanks to @zainab-amir )
> 
> If we use this approach for hiding the social auth skeleton when they are inactive, then it will also hide the skeleton when social auth is active. Because we get providers in MFE_context api response and based on the response we render these buttons if providers are available.

When social auth is active - skeleton will not be hidden. I recorded a video with local check (sorry for quality, I need to made it less than 10mb)

Uploading Запись экрана 2023-10-21 в 14.09.23-2.mov…

**Comment by mphilbrick211** [comment]
Hi @Lunyachek! Would you mind rebasing to see if that fixes the codecov failure?

Also, @zainab-amir just flagging this for review, along with the [backport](https://github.com/openedx/frontend-app-authn/pull/766).

**Comment by syedsajjadkazmii** [comment]
Hi @Lunyachek 
>I agree, changed to isThirdPartyAuthActive
please change all `ThirdPartyAuthPreloader` to `isThirdPartyAuthActive`. 

also i think you can use `isThirdPartyAuthActive` variable instead of condition in the following place
[login page component](https://github.com/openedx/frontend-app-authn/pull/767/files#diff-fe27103c5d8359e134c308a53df0dd186957e353602933817199849936ec6c61R173) and 
[third party auth component](https://github.com/openedx/frontend-app-authn/pull/767/files#diff-3c933b31d43a90947e2d31a61635cc743112e0533f5d966a39c4193548b3fe68R32) 


And please rebase the branch :)

**Comment by mphilbrick211** [comment]
> Hi @Lunyachek, you are right. The loader is not hidden when social auth is enabled.
> 
> But could you please do the following changes?
> 
> > I agree, changed to isThirdPartyAuthActive
> 
> please change all `ThirdPartyAuthPreloader` to `isThirdPartyAuthActive`.
> 
> also i think you can use `isThirdPartyAuthActive` variable instead of condition in the following places [login page component](https://github.com/openedx/frontend-app-authn/pull/767/files#diff-fe27103c5d8359e134c308a53df0dd186957e353602933817199849936ec6c61R173) and [third party auth component](https://github.com/openedx/frontend-app-authn/pull/767/files#diff-3c933b31d43a90947e2d31a61635cc743112e0533f5d966a39c4193548b3fe68R32)
> 
> And please rebase the branch :)

Hi @Lunyachek! Flagging this for you.

**Comment by dyudyunov** [comment]
@mphilbrick211 hi

I rebased the branch, now we need someone to trigger workflows 

@Lunyachek please check recent comments:

>Hi @Lunyachek, you are right. The loader is not hidden when social auth is enabled.
But could you please do the following changes?
I agree, changed to isThirdPartyAuthActive
please change all ThirdPartyAuthPreloader to isThirdPartyAuthActive.
also i think you can use isThirdPartyAuthActive variable instead of condition in the following places
[login page component](https://github.com/openedx/frontend-app-authn/pull/767/files#diff-fe27103c5d8359e134c308a53df0dd186957e353602933817199849936ec6c61R173) and
[third party auth component](https://github.com/openedx/frontend-app-authn/pull/767/files#diff-3c933b31d43a90947e2d31a61635cc743112e0533f5d966a39c4193548b3fe68R32)

**Comment by mphilbrick211** [comment]
@Lunyachek @dyudyunov - tests have been re-run. Looks like there's still a failing check.

**Comment by mphilbrick211** [comment]
> @Lunyachek @dyudyunov - tests have been re-run. Looks like there's still a failing check.

Hi @Lunyachek and @dyudyunov - it would be great to get this reviewed soon. Flagging the failing check and the branch conflicts that have popped up.

**Comment by dyudyunov** [comment]
@mphilbrick211 thanks for the ping!
We'll try to work on it ASAP

Note: codecov (failed check) is pointing to lines where no changes were made. The rebasing might fix it

**Comment by Lunyachek** [comment]
I rebased branch and moved changes to the ThirdPartyAuth.jsx

**Comment by mphilbrick211** [comment]
Hi @openedx/2u-vanguards! This is ready for review, along with backports https://github.com/openedx/frontend-app-authn/pull/766 and https://github.com/openedx/frontend-app-authn/pull/1225

**Review by syedsajjadkazmii — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@Lunyachek 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-authn&entry.867055334=https://github.com/openedx/frontend-app-authn/pull/767&entry.1484655318=https://github.com/Lunyachek&entry.752974735=2023-03-01+15:15&entry.1917517419=2024-04-19+05:55&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**